crates/ruff/src/message/gitlab.rs

```diff
@@ -58,6 +57,7 @@ impl Serialize for SerializedMessages<'_> {
         S: Serializer,
     {
         let mut s = serializer.serialize_seq(Some(self.messages.len()))?;
+        let mut fingerprints = HashSet::<String>::new();
```

Comment:
We can already initialize the hash set with the right capacity to avoid multiple resizes (and, copying over all elements multiple time). 

```suggestion
        let mut fingerprints = HashSet::<String>::with_capacity(self.messages.len());
```